### PR TITLE
Add mutation to create quote with onboarding session

### PR DESCRIPTION
**Pull request by tobiasbexelius**
# Jira Issue: [MX-248] 

## What?
- Add mutation to create quote with onboarding session instead of member
- Remove deprecated quote types `apartment` and `house`

## Why?
- 

## Optional checklist
- [x] Codescouted
- [x] Unit tests written
- [x] Tested locally

[MX-248]: https://hedvig.atlassian.net/browse/MX-248

### CODE REVIEW

**Review by mightyost — COMMENTED** [review]
Some cosmetic comments except for `house` and `apartment` that cannot be removed yet, right?

I also thinks it would be a good idea to store the `OnboardingSession` id in the `Quote` and also persist it in db, would give us:

- traceability
- nicer code, it does not have to be feed into every strategy to get it into the notification event

File: src/main/kotlin/com/hedvig/underwriter/service/QuoteServiceImpl.kt [hunk 1]
```diff
@@ -155,6 +157,44 @@ class QuoteServiceImpl(
             .map((QuoteDto)::fromQuote)
 
     override fun createQuote(
+        id: UUID?,
+        quoteRequest: QuoteRequest,
+        initiatedFrom: QuoteInitiatedFrom,
+        underwritingGuidelinesBypassedBy: String?
+    ): Either<ErrorResponseDto, CompleteQuoteResponseDto> {
```
[the remark this answers is not in the document]
Comment (tobiasbexelius) [reply src/main/kotlin/com/hedvig/underwriter/service/QuoteServiceImpl.kt:164]: The idea is to remove createAndOnboardMember once we have moved to not creating members until sign. But I should add a comment

File: src/main/kotlin/com/hedvig/underwriter/graphql/Mutation.kt [hunk 2]
```diff
@@ -84,6 +87,70 @@ class Mutation @Autowired constructor(
         }
     }
 
+    @LogCall
+    fun createQuoteV2(input: CreateQuoteInput, env: DataFetchingEnvironment): CreateQuoteResult {
+        val cleanInput = cleanInput(input)
```
[the remark this answers is not in the document]
Comment (tobiasbexelius) [reply src/main/kotlin/com/hedvig/underwriter/graphql/Mutation.kt:92]: Makes sense! Our strategy with the memberless onboarding was to not mess with the existing code, but just create separate versions instead, and remove the old ones later. But this turned our more similar than expected, so I don't think we'll need a v2.

File: src/main/kotlin/com/hedvig/underwriter/service/QuoteServiceImpl.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (tobiasbexelius) [reply src/main/kotlin/com/hedvig/underwriter/service/QuoteServiceImpl.kt:164]: Actually I take that back.

File: src/main/kotlin/com/hedvig/underwriter/graphql/Mutation.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (tobiasbexelius) [reply src/main/kotlin/com/hedvig/underwriter/graphql/Mutation.kt:92]: After looking into merging, I do prefer to keep them separate. That makes it easier to iterate on the new flow.

File: src/main/kotlin/com/hedvig/underwriter/service/QuoteServiceImpl.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (tobiasbexelius) [reply src/main/kotlin/com/hedvig/underwriter/service/QuoteServiceImpl.kt:164]: I take back my takeback

File: src/main/kotlin/com/hedvig/underwriter/graphql/Mutation.kt [hunk 3]
```diff
@@ -43,8 +47,7 @@ class Mutation @Autowired constructor(
     fun createQuote(createQuoteInput: CreateQuoteInput, env: DataFetchingEnvironment): CreateQuoteResult {
 
         val input = when {
-            createQuoteInput.apartment != null || createQuoteInput.house != null ||
-                createQuoteInput.swedishApartment != null || createQuoteInput.swedishHouse != null -> {
```
[the remark this answers is not in the document]
Comment (tobiasbexelius) [reply src/main/kotlin/com/hedvig/underwriter/graphql/Mutation.kt:47]: Oh, good catch! That must be a partner, I could not find any usages in our github, but I should have checked datadog instead for sure! Will investigate

File: src/main/kotlin/com/hedvig/underwriter/graphql/Mutation.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (mightyost) [reply src/main/kotlin/com/hedvig/underwriter/graphql/Mutation.kt:92]: Oki, understand :), just make sure to remove old logic later ;).

:)!

File: src/main/kotlin/com/hedvig/underwriter/graphql/Mutation.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (tobiasbexelius) [reply src/main/kotlin/com/hedvig/underwriter/graphql/Mutation.kt]: Reverted for now

**Review by mightyost — APPROVED** [review]
👍 

I still thinks it would be a good idea to add the `onboardingSessionId` to the `Quote` and have it stored in the `quote_revisions` table for traceability. As a side effect it will also be available in all Guidelines without you to have to feed it in on the side as now...

**Comment by tobiasbexelius** [comment]
@mightyost Agree. Will definitely keep that in mind, but for another PR

**Pull request merged**